[29기 문지원] westagram login/main

moonkorea00 opened the pull request:
## :: Westagram login/main page
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표
- 로그인 기능
- 로그인 입력값에 따른 버튼 활성화 기능 (js)
- 댓글 기능 구현
- 좋아요 / 삭제 기능

<br />

## :: 구현 사항 설명 
1. input 창에 addEventListener로 함수 등록
2. eventListener안에 입력값을 매번 받아서 검사
3. 검사값이 true면 버튼의 색상을 파란색으로 변경
4. false면 버튼을 불투명하게 변경

<br />

## :: 성장 포인트
- html에서의 요소 작성 후 css style 작성
- 
- -semantic 태그의 사용을  습관화하며 클래스와 아이디 작명시 semantic하게 작성. 특히 요소가 많을수록 복잡
<img width="1088" alt="Screen Shot 2022-01-07 at 5 20 56 PM" src="https://user-images.githubusercontent.com/78708082/148513951-2f552c14-c23d-40a9-ab80-c0feaea478bc.png">

- css속성은 레이아웃에 영향을 많이 주는 순서와 인접한 속성대로 작성

- 레이아웃을 만들때 Margin, Position 등에 값을 부여하여 수평/수직 정렬 기능을 구현하기 보다는 CSS flexbox 모듈로 통일하여 기능을 부여하는 것이 유지/보수 관점에서도 바람직함

- 부모요소에 높이/너비를 부여하면 요소 추가/수정을 할때 유지보수가 어려움

- 들여쓰기는 beautify 

-  westagram main.html에 네비게이션바(navbar)를 구현하는데 display: flex, justify-content, align-items를 자식 요소에도 부여를 해야 목표하는 수평정렬이 된다는 점을 20시간 넘게 투자하며 의도치 않게 어렵게 배움


<br />

## :: 기타 질문 및 특이 사항
- 특이사항으로는 html css 레이아웃(미완성)만 학습/구현하는데 60시간 이상 투자

joonsikyang commented on students/15th/HaiinKim/js/login.js:
```diff
@@ -0,0 +1,17 @@
+const inputPassword = document.querySelector('.loginPasswordInput')
+const btn = document.querySelector('.btn')
+
+inputPassword.addEventListener('keyup', moreThanOne)
```
이렇게 하면 패스워드를 먼저 작성하는 경우에는 동작하지 않겠네요! 어떻게 하는게 좋을까요? :)

Haiin05 replied:
id, password 모두 적용하여 수정하였습니다.

joonsikyang commented on students/15th/HaiinKim/js/main.js:
```diff
@@ -0,0 +1,44 @@
+//함수 분리하기
+
+
+const form = document.querySelector('.mainFeedCommentBox')
+const input = document.querySelector('.mainFeedCommentBoxInput');
+const id = document.querySelector('.mainFeedContent_name');
+const ul = document.querySelector('.mainFeedComments');
+console.log(ul)
+const li = document.createElement('li');
+li.classList.add('new_comment');
+const span1 = document.createElement('span');
+const span2 = document.createElement('span');
+span1.className = 'comment_user';
+span2.className = 'comment_comm';
```
변수 선언과 로직 실행(ex. className 부여)을 분리하면 좋겠네요. 변수는 필요한 요소를 가져오고, 특정 상황에 필요한 로직이 실행될 수 있도록 함수로 분리해주세요! 위에 주석이 그 뜻인가요..?

Haiin05 replied:
변수 선언과 로직 분리하고 이름도 수정하였습니다.

joonsikyang commented on students/15th/HaiinKim/js/main.js:
```diff
@@ -0,0 +1,44 @@
+form.addEventListener('submit', newComment)
+
+
+function newComment (event) {
+    event.preventDefault();
+    span1.innerText = 'Haiin.k';
+    span2.innerHTML = input.value;
+    li.append(span1);
+    li.append(span2);
+    ul.appendChild(li);
+    input.value = "";
+} 
```
- newComment 보다는 addComment와 같은 함수명이 더 좋겠네요! 보통 동사형으로 만듭니다.
- innerHTML 메소드를 좀 더 잘 사용해보면 코드가 훨씬 깔끔해질 수 있겠네요 :) 아래 참고해주세요.
```
ul.innerHTML = `
    <li>
        <span>Haiin.k</span>
    <li>
`
```

Haiin05 replied:
이렇게 고쳐봤는데 다른 리스트를 덮어쓰거나  코드가 아예 실행이 안되서 정리만 했습니다. 오프라인으로 여쭤보겠습니다.

joonsikyang commented on students/15th/HaiinKim/js/main.js:
```diff
@@ -0,0 +1,44 @@
+    ul.appendChild(li);
+    input.value = "";
+} 
+// function userSpan () {
+//     const userNameSpan = document.createElement('span');
+//     const list = document.createElement('li');
+//     list.setAttribute('class', 'newComment');
+//     list.appendChild(userNameSpan);
+
+
+   
+    // if (comment.length >= 1) {
+    //     btn.style.color = "rgb(44, 134, 224)"
+    // };
```
불필요한 코드는 삭제해주세요!

Haiin05 replied:
네! 정리했습니다.

joonsikyang commented on students/15th/HaiinKim/login.html:
```diff
@@ -0,0 +1,27 @@
+    </head>
+    <body>
+        <div class="frame">
+
```
보통 html 요소와 요소 사이에 한 줄 띄우지 않습니다!

Haiin05 replied:
네 정리했습니다.

joonsikyang commented on students/15th/HaiinKim/main.html:
```diff
@@ -0,0 +1,223 @@
+                            <i class="fas fa-ellipsis-h" style="color:rgba(0, 0, 0, 0.7)"></i>    
+                        </div>
+                        <div class="mainFeedPic">
+                            <img src="santiago.JPG">
```
img 태그에 alt 값 반드시 부여해주세요! SEO와 관련있습니다. img 태그 전반적으로 검토 부탁드립니다!

Haiin05 replied:
alt개념 알게되었습니다. 수정하였습니다.

joonsikyang reviewed (commented):
코드가 깔끔하네요🙌 리뷰 확인해주세요-!
> 클래스 네이밍이 너무 긴 것 같아 다 구현하고 짧게 고쳐야 하나 생각하고 있습니다.
- 구현하고 고치면 이미 늦습니다.
- 지금은 조금 길어도 나중에 React 배우면서 Sass 적용하면 훨씬 효율적으로 클래스명을 부여할 수 있습니다. 현재는 조금 길더라도 명확한 변수명, 함수명, 클래스명 사용하는 것에 더 신경쓰면 됩니다! 괜찮아요~

Haiin05 commented on students/15th/HaiinKim/README.md:
```diff
@@ -0,0 +1,2 @@
+# wecode
```
위코드! :)

claire-euni commented:
closed

moonkorea00 committed:
-버튼은 이렇게 했다
 - 이렇게 했다
 with '#' will be ignored, and an empty message aborts the commit.

moonkorea00 committed:
"2차 테스트"

moonkorea00 committed:
추가 예정 사항:
 - 댓글 기능 구현
 - main-container hmtl과 css 구현
 - navbar를 제외한 feeds 컨테이너 css 구현
 - search-box에 돋보기도 placeholder로 구현 되게 변경
 - 댓글 기능 구현 전 self refactoring 실시
 - repository에login.html과 main.html commit
 - PR 메시지와 라벨 등으로 소통 및 기록

Westagram login html/css/js
  HTML 작성
  -로그인 전체를 감싸고 border가 적용된 div 태그 사용
  -instagram 폰트 사용
  -input 태그 사용
  -button 태그 사용

 구현사항
  -id 및 pw에 각각 한 글자 이상 입력했을때 button 활성화
  -활성화되면 button.style.backgrounColor 변경

moonkorea00 committed:
로그인 html css js

moonkorea00 committed:
로그인 html, css, js 복구

moonkorea00 committed:
로그인 html css js 복구
